### PR TITLE
libid3tag: fix arm64 build

### PR DESCRIPTION
**Pull request by osheroff**
this approach will/should fix any package that gets "config.guess/ config.sub" errors during configure. In some ways I'd rather just copy the files wholesale instead of patching, but I couldn't figure out an elegant way to do that within a formula.

### CODE REVIEW

**Comment by carlocab** [comment]
What is the source of this patch? Has it been sent upstream? When can it be removed?

I think these will all need to be documented in a comment before the patch.

**Comment by fxcoudert** [comment]
Can we instead add automake/autoconf/libtool build dependencies, and run autoreconf?

**Comment by osheroff** [comment]
> What is the source of this patch? Has it been sent upstream? When can it be removed?

upstream (https://github.com/markjeee/libid3tag) has been dead for 11 years now.  I figure they're not coming back.

> Can we instead add automake/autoconf/libtool build dependencies, and run autoreconf?

ah yeah that's better, I'll try that.

**Comment by osheroff** [comment]
updated

File: Formula/libid3tag.rb [hunk 1]
```diff
@@ -66,7 +66,13 @@ class Libid3tag < Formula
     sha256 "5e86270ebb179d82acee686700d203e90f42e82beeed455b0163d8611657d395"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+
   def install
+    system "touch NEWS AUTHORS ChangeLog"
```
Comment (fxcoudert) [inline Formula/libid3tag.rb:74]: Any reason why this is necessary?

Comment (fxcoudert) [reply Formula/libid3tag.rb:74]:
```suggestion
    system "touch", "NEWS", "AUTHORS", "ChangeLog"
```

File: Formula/libid3tag.rb [hunk 2]
```diff
@@ -66,7 +66,13 @@ class Libid3tag < Formula
...
+    system "autoreconf --install -f"
```
Comment (fxcoudert) [inline Formula/libid3tag.rb:75]:
```suggestion
    system "autoreconf", "-fiv"
```

File: Formula/libid3tag.rb [hunk 3]
```diff
@@ -66,12 +66,13 @@ class Libid3tag < Formula
     sha256 "5e86270ebb179d82acee686700d203e90f42e82beeed455b0163d8611657d395"
   end
 
-  patch do
-    url "https://gist.githubusercontent.com/osheroff/afa7452c4512e5d045564f7cca37155f/raw/6c668888b01818865c1dc1ad6c58eeb2518116af/libid3tag_autoconf.patch"
-    sha256 "205df514ff15ab5f1d472d5603d4657e7eec38a629e7a77c2c4720565c5c7f5e"
-  end
+  depends_on "autoconf" => :build
```
Comment (carlocab) [inline Formula/libid3tag.rb:69]: You should move this up to before `uses_from_macos "zlib"`. Leave a space in between.

**Comment by carlocab** [comment]
I think this is

    license "GPL-2.0-only"

Sources: 
1. https://www.underbit.com/resources/license/gpl
1. `grep`ping the source tarball only returns references to the `COPYING` file

File: Formula/libid3tag.rb (hunk 1 above)
Comment (osheroff) [reply Formula/libid3tag.rb:74]: without it: 

```
Makefile.am: error: required file './NEWS' not found
Makefile.am: error: required file './AUTHORS' not found
Makefile.am: error: required file './ChangeLog' not found
```

You can sorta around it by doing like:

"AUTOMAKE='automake --foreign' autoreconf" 

but I figured this approach was cleaner on some level

File: Formula/libid3tag.rb (hunk 3 above)
Comment (osheroff) [reply Formula/libid3tag.rb]: fixed

**Comment by carlocab** [comment]
you can add the license immediately after the `sha256 ".*"` line.

**Comment by osheroff** [comment]
updated again.

**Comment by osheroff** [comment]
ping, should be good to go?

**Comment by osheroff** [comment]
oh right i forgot it's the weekend and that weekends still exist

**Review by fxcoudert — APPROVED** [review]

**Pull request closed without merging**